packages/common/src/api/tan-query/useNotifications.ts

```diff
@@ -0,0 +1,235 @@
+    ) {
+      if (notification.entityType === Entity.Track) {
+        trackIds.add(notification.entityId)
+      }
+      notification.userIds.forEach((id) => userIds.add(id))
+    }
+  })
+
+  return { userIds, trackIds, collectionIds }
+}
+
+/**
+ * Hook that returns paginated notifications for the current user.
+ * Uses infinite query to support "Load More" functionality.
+ * Pagination is based on the timestamp and groupId of the last notification.
+ */
+export const useNotifications = (options?: Config) => {
+  const { audiusSdk } = useAudiusQueryContext()
+  const { data: currentUserId } = useCurrentUserId()
+  const validTypes = useNotificationValidTypes()
+  const query = useInfiniteQuery({
+    queryKey: [QUERY_KEYS.NOTIFICATIONS, currentUserId],
+    initialPageParam: null as PageParam,
+    queryFn: async ({ pageParam = null }) => {
+      const sdk = await audiusSdk()
+      const { data } = await sdk.full.notifications.getNotifications({
+        userId: Id.parse(currentUserId),
+        limit: DEFAULT_LIMIT,
+        timestamp: (pageParam as PageParam)?.timestamp,
+        groupId: (pageParam as PageParam)?.groupId,
+        validTypes
+      })
+
+      return transformAndCleanList(data?.notifications, notificationFromSDK)
+    },
+    getNextPageParam: (lastPage: Notification[]) => {
+      const lastNotification = lastPage[lastPage.length - 1]
+      if (!lastNotification || lastPage.length < DEFAULT_LIMIT) {
+        return null
+      }
+      return {
+        timestamp: lastNotification.timestamp,
+        groupId: lastNotification.groupId
+      }
+    },
+    ...options,
+    enabled: options?.enabled !== false && !!currentUserId
+  })
+
+  const lastPage = query.data?.pages[query.data.pages.length - 1]
+  const { userIds, trackIds, collectionIds } = lastPage
+    ? collectEntityIds(lastPage)
+    : {
+        userIds: new Set<ID>(),
+        trackIds: new Set<ID>(),
+        collectionIds: new Set<ID>()
+      }
+
+  // Pre-fetch related entities
+  const usersQuery = useUsers(Array.from(userIds))
+  const tracksQuery = useTracks(Array.from(trackIds))
+  const collectionsQuery = useCollections(Array.from(collectionIds))
+
+  // Check if the latest page's entity data is still loading
+  const isLatestPagePending =
+    usersQuery.isPending || tracksQuery.isPending || collectionsQuery.isPending
+
+  const isLatestPageLoading =
+    usersQuery.isLoading || tracksQuery.isLoading || collectionsQuery.isLoading
+
+  const isError =
+    query.isError ||
+    usersQuery.isError ||
+    tracksQuery.isError ||
+    collectionsQuery.isError
+
+  // Return all pages except the last one if it's still loading entity data
+  const notifications = query.data?.pages.slice(0, -1).flat() ?? []
+  if (!isLatestPagePending && lastPage) {
+    notifications.push(...lastPage)
+  }
+
+  return {
+    ...query,
+    isPending: query.isPending || isLatestPagePending,
+    isLoading: query.isLoading || isLatestPageLoading,
```

Comment:
Not sure about overriding these. I guess it's necessary in order to block on the prefetching. If we are doing this, we should also technically hit `status` and the other `is` bools if we're really trying to maintain the query results contract.

Reply:
if we do go this way, we should make utils to make it easier. Something that merges multiple query results and gives an overall status and related bools so we can reuse it

Reply:
yeah totally agreed, adding this as a topic for standup tomorrow